metricsaggregator/src/test/java/io/dockstore/metricsaggregator/MetricsAggregatorS3ClientTest.java

```diff
@@ -0,0 +1,142 @@
+import io.dockstore.webservice.core.Partner;
+import io.dockstore.webservice.helpers.S3ClientHelper;
+import io.dropwizard.testing.DropwizardTestSupport;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest;
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+import static io.dockstore.client.cli.BaseIT.ADMIN_USERNAME;
+import static io.dockstore.metricsaggregator.common.TestUtilities.BUCKET_NAME;
+import static io.dockstore.metricsaggregator.common.TestUtilities.ENDPOINT_OVERRIDE;
+import static io.dockstore.metricsaggregator.common.TestUtilities.createExecution;
+import static io.dockstore.openapi.client.model.Execution.ExecutionStatusEnum.SUCCESSFUL;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(LocalstackDockerExtension.class)
+@LocalstackDockerProperties(imageTag = LocalStackTestUtilities.IMAGE_TAG, services = { ServiceName.S3 }, environmentVariableProvider = LocalStackTestUtilities.LocalStackEnvironmentVariables.class)
+class MetricsAggregatorS3ClientTest {
+    private static MetricsAggregatorS3Client metricsAggregatorS3Client;
+    private static S3Client s3Client;
+    private static TestingPostgres testingPostgres;
+
+    public static DropwizardTestSupport<DockstoreWebserviceConfiguration> SUPPORT = new DropwizardTestSupport<>(
+            DockstoreWebserviceApplication.class, CommonTestUtilities.PUBLIC_CONFIG_PATH);
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        CommonTestUtilities.dropAndRecreateNoTestData(SUPPORT, CommonTestUtilities.PUBLIC_CONFIG_PATH);
+        SUPPORT.before();
+        testingPostgres = new TestingPostgres(SUPPORT);
+
+        metricsAggregatorS3Client = new MetricsAggregatorS3Client(BUCKET_NAME, ENDPOINT_OVERRIDE);
+        s3Client = TestUtils.getClientS3V2(); // Use localstack S3Client
+        // Create a bucket to be used for tests
+        CreateBucketRequest request = CreateBucketRequest.builder().bucket(BUCKET_NAME).build();
+        s3Client.createBucket(request);
+        deleteBucketContents(); // This is here just in case a test was stopped before tearDown could clean up the bucket
+    }
+
+    @BeforeEach
+    public void dropAndRecreateDB() {
+        CommonTestUtilities.dropAndCreateWithTestDataAndAdditionalToolsAndWorkflows(SUPPORT, false, CommonTestUtilities.PUBLIC_CONFIG_PATH);
```

Comment:
Seems redundant with line 76, same explanation as the other PR

Reply:
Removing line 76 results in a schema validation error: `org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: missing table [apptool]`